contracts/ProofOfHumanity.sol

```diff
@@ -486,8 +498,7 @@ contract ProofOfHumanity is IArbitrable, IEvidence {
         for (uint i = 0; i<_vouches.length && request.vouches.length<requiredNumberOfVouches; i++) {
             // Check that the vouch isn't currently used by another submission and the voucher has a right to vouch.
             Submission storage voucher = submissions[_vouches[i]];
-            if (!voucher.hasVouched && voucher.registered && now - voucher.submissionTime <= submissionDuration &&
```

Comment:
@unknownunknown1 I think we should keep this, taking into account that a vouch could have been added a long time ago at this point. Maybe the person was registered at the time of vouching but then got removed.

Reply:
Yeah, you're probably right. The vouch can be old and will carry to reapply submission, though it might be irrelevant in the present

Reply:
I can't add it apparently because of the size limit though I agree that we need this line.
I'd rather remove AppealContribution event since it's not really a necessity. But maybe you have some other ideas

Reply:
Use codes for error messages.

Reply:
Every error message except maybe 1 already fits into 32 bytes size. Will using the codes make any difference in this case?

I deleted the event for now

Reply:
Maybe you can try reducing runs in the optimizer config https://github.com/Proof-Of-Humanity/Proof-Of-Humanity/blob/d763c0a220e144c0e5da42be5822d5d79a4b9bfd/truffle-config.js#L10. If it works, a drawback could be worse gas efficiency though.

Reply:
Making the onlyGovernor modifier an internal function?

Reply:
I'm not sure that trading this event for gas efficiency is the best solution. I'll try it with modifier though.
Thing is that in previous commit the event wasn't implemented properly anyway. It was missing _challengeID which I couldn't add, and as you said yourself it should be emitted when 0 round is funded as well.
We can always move view functions to a different contract, but this particular case doesn't seem worth it. It can be discussed though

Reply:
Replacing the modifier helped, thanks. I also shorten one of the require messages, so they all fit in 32 bytes now.
I updated the code. Please, check

Reply:
Looks good now!
I'm surprised that using an internal function like that worked though. I was thinking of placing the entire require statement inside the isGovernor() function, so that the error string wouldn't be repeated many times. I expected that string repetition to be the problem

Reply:
I think the compiler stores the string in one place.